GitUI/Avatars/AvatarService.cs

```diff
@@ -1,12 +1,126 @@
-﻿using GitUI.Properties;
+﻿using System;
+using System.Linq;
+using GitCommands;
+using GitUI.Properties;
 
 namespace GitUI.Avatars
 {
     public static class AvatarService
     {
-        private static readonly InitialsAvatarGenerator AvatarGenerator = new InitialsAvatarGenerator();
-        public static IAvatarProvider Default { get; }
-            = new BackupAvatarProvider(new AvatarMemoryCache(new AvatarPersistentCache(new AvatarDownloader(AvatarGenerator), AvatarGenerator)),
-                Images.User80);
+        private static readonly InitialsAvatarProvider InitialsAvatarProvider;
+        private static readonly StaticImageAvatarProvider UserImageAvatarProvider;
+
+        private static readonly HotSwapAvatarProvider HotSwapProvider;
```

Comment:
We have IAvatarProvider but implementation dependent.
Please change to IAvatarProvider.

Reply:
those fields are private and there is no harm in keeping the type as concrete as possible.

Reply:
Try to change it, if everything is ok, then the code will work and you have not violated the contract.

Reply:
I could change the first two because it doesn't matter internally. The HotSwapAvatarProvider can't be changed, because it would break `UpdateAvatarProvider()` - This method is used to set the new inner provider (depending on the settings) and depends on the `HotSwapAvatarProvider.Provider` property.

I'm also not sure what contract I'm violating, because the class `AvatarService` doesn't have any interfaces (and also couldn't because it's static) and the fields are private and only an implementation detail of `AvatarService` - I can't see any contracts here.